### PR TITLE
(internal): change host from 0.0.0.0 to localhost

### PR DESCRIPTION
**Pull request by erwinheitzman**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by erwinheitzman** [comment]
@christian-bromann I cannot say for sure but it seems like the devtools service is the only thing blocking this from passing. Making these changes locally seems like it does not produce issues when running the same setup (devtools) like we use in the tests, therefore I think it might be the tests and not the code but I'm not confident about this at all and appreciate any help.

**Comment by christian-bromann** [comment]
> it seems like the devtools service is the only thing blocking this from passing.

I am seeing some unit tests failing as well. I am assuming of these pass the e2e test will fail as of my understanding these settings are needed. I could be wrong though.

**Comment by erwinheitzman** [comment]
Yeah the unit tests that are failing are related to the devtools/puppeteer as well

**Comment by christian-bromann** [comment]
I don't think these tests are related to Puppeteer implementation:
```
⎯⎯⎯⎯⎯⎯⎯ Failed Tests 2 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  packages/wdio-utils/tests/node/index.test.ts > startWebDriver > should start firefox driver
AssertionError: expected "spy" to be called with arguments: [ { port: 1234, foo: 'bar', …(1) } ]

Received: 

  1st spy call:

  Array [
    Object {
-     "cacheDir": Any<String>,
+     "allowHosts": Array [
+       "localhost",
+     ],
+     "cacheDir": "/tmp",
      "foo": "bar",
      "port": 1234,
    },
  ]


Number of calls: 1

 ❯ packages/wdio-utils/tests/node/index.test.ts:156:34
    154|         })
    155|         expect(startGeckodriver).toBeCalledTimes(1)
    156|         expect(startGeckodriver).toBeCalledWith({
       |                                  ^
    157|             port: 1234,
    158|             foo: 'bar',

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/2]⎯

 FAIL  packages/wdio-utils/tests/node/index.test.ts > startWebDriver > should start no driver or download geckodriver if binaries are defined
AssertionError: expected "spy" to be called with arguments: [ { cacheDir: Any{ …(3) }, …(2) } ]

Received: 

  1st spy call:

  Array [
    Object {
-     "cacheDir": Any<String>,
+     "allowHosts": Array [
+       "localhost",
+     ],
+     "cacheDir": "/tmp",
      "customGeckoDriverPath": "/my/geckodriver",
      "port": 1234,
    },
  ]
```

**Comment by erwinheitzman** [comment]
Oh those did not fail for me locally, I can check those when I have time to look into this again

**Comment by christian-bromann** [comment]
@erwinheitzman it seems like component tests are passing so the driver actually can open on `localhost`. The issue in build seems to be connected with Puppeteer not able to connect 🤔

File: website/i18n/fa/docusaurus-plugin-content-docs/current/Proxy.md [hunk 1]
```diff
@@ -81,7 +81,7 @@ export const config = {
             httpProxy: "corporate.proxy:8080",
             socksUsername: "codeceptjs",
             socksPassword: "secret",
-            noProxy: "127.0.0.1,localhost"
+            noProxy: "localhost,localhost"
```
Comment (christian-bromann) [inline website/i18n/fa/docusaurus-plugin-content-docs/current/Proxy.md:84]:
```suggestion
            noProxy: "localhost"
```

File: website/i18n/fr/docusaurus-plugin-content-docs/current/Proxy.md [hunk 2]
```diff
@@ -81,7 +81,7 @@ export const config = {
             httpProxy: "corporate.proxy:8080",
             socksUsername: "codeceptjs",
             socksPassword: "secret",
-            noProxy: "127.0.0.1,localhost"
+            noProxy: "localhost,localhost"
```
Comment (christian-bromann) [inline website/i18n/fr/docusaurus-plugin-content-docs/current/Proxy.md:84]:
```suggestion
            noProxy: "localhost"
```

File: website/i18n/hi/docusaurus-plugin-content-docs/current/Proxy.md [hunk 3]
```diff
@@ -81,7 +81,7 @@ sc -u $SAUCE_USERNAME -k $SAUCE_ACCESS_KEY --no-autodetect -p http://my.corp.pro
             httpProxy: "corporate.proxy:8080",
             socksUsername: "codeceptjs",
             socksPassword: "secret",
-            noProxy: "127.0.0.1,localhost"
+            noProxy: "localhost,localhost"
```
Comment (christian-bromann) [inline website/i18n/hi/docusaurus-plugin-content-docs/current/Proxy.md:84]:
```suggestion
            noProxy: "localhost"
```

File: website/i18n/it/docusaurus-plugin-content-docs/current/Proxy.md [hunk 4]
```diff
@@ -81,7 +81,7 @@ export const config = {
             httpProxy: "corporate.proxy:8080",
             socksUsername: "codeceptjs",
             socksPassword: "secret",
-            noProxy: "127.0.0.1,localhost"
+            noProxy: "localhost,localhost"
```
Comment (christian-bromann) [inline website/i18n/it/docusaurus-plugin-content-docs/current/Proxy.md:84]:
```suggestion
            noProxy: "localhost"
```

File: website/i18n/ja/docusaurus-plugin-content-docs/current/Proxy.md [hunk 5]
```diff
@@ -81,7 +81,7 @@ export const config = {
             httpProxy: "corporate.proxy:8080",
             socksUsername: "codeceptjs",
             socksPassword: "secret",
-            noProxy: "127.0.0.1,localhost"
+            noProxy: "localhost,localhost"
```
Comment (christian-bromann) [inline website/i18n/ja/docusaurus-plugin-content-docs/current/Proxy.md:84]:
```suggestion
            noProxy: "localhost"
```

File: website/i18n/pl/docusaurus-plugin-content-docs/current/Proxy.md [hunk 6]
```diff
@@ -81,7 +81,7 @@ export const config = {
             httpProxy: "corporate.proxy:8080",
             socksUsername: "codeceptjs",
             socksPassword: "secret",
-            noProxy: "127.0.0.1,localhost"
+            noProxy: "localhost,localhost"
```
Comment (christian-bromann) [inline website/i18n/pl/docusaurus-plugin-content-docs/current/Proxy.md:84]:
```suggestion
            noProxy: "localhost"
```

File: website/i18n/ru/docusaurus-plugin-content-docs/current/Proxy.md [hunk 7]
```diff
@@ -81,7 +81,7 @@ export const config = {
             httpProxy: "corporate.proxy:8080",
             socksUsername: "codeceptjs",
             socksPassword: "secret",
-            noProxy: "127.0.0.1,localhost"
+            noProxy: "localhost,localhost"
```
Comment (christian-bromann) [inline website/i18n/ru/docusaurus-plugin-content-docs/current/Proxy.md:84]:
```suggestion
            noProxy: "localhost"
```

File: website/i18n/ta/docusaurus-plugin-content-docs/current/Proxy.md [hunk 8]
```diff
@@ -81,7 +81,7 @@ export const config = {
             httpProxy: "corporate.proxy:8080",
             socksUsername: "codeceptjs",
             socksPassword: "secret",
-            noProxy: "127.0.0.1,localhost"
+            noProxy: "localhost,localhost"
```
Comment (christian-bromann) [inline website/i18n/ta/docusaurus-plugin-content-docs/current/Proxy.md:84]:
```suggestion
            noProxy: "localhost"
```

File: website/i18n/uk/docusaurus-plugin-content-docs/current/Proxy.md [hunk 9]
```diff
@@ -81,7 +81,7 @@ export const config = {
             httpProxy: "corporate.proxy:8080",
             socksUsername: "codeceptjs",
             socksPassword: "secret",
-            noProxy: "127.0.0.1,localhost"
+            noProxy: "localhost,localhost"
```
Comment (christian-bromann) [inline website/i18n/uk/docusaurus-plugin-content-docs/current/Proxy.md:84]:
```suggestion
            noProxy: "localhost"
```

File: website/i18n/zh/docusaurus-plugin-content-docs/current/Proxy.md [hunk 10]
```diff
@@ -81,7 +81,7 @@ export const config = {
             httpProxy: "corporate.proxy:8080",
             socksUsername: "codeceptjs",
             socksPassword: "secret",
-            noProxy: "127.0.0.1,localhost"
+            noProxy: "localhost,localhost"
```
Comment (christian-bromann) [inline website/i18n/zh/docusaurus-plugin-content-docs/current/Proxy.md:84]:
```suggestion
            noProxy: "localhost"
```

File: packages/webdriver/src/utils.ts [hunk 11]
```diff
@@ -419,8 +419,7 @@ function getExecCmdArgs(requestOptions: Options.RequestLibOptions): string {
  * @param socketUrl url to bidi interface
  * @returns prototype with interface for bidi primitives
  */
-export function initiateBidi (socketUrl: string, strictSSL: boolean = true): PropertyDescriptorMap {
-    socketUrl = socketUrl.replace('localhost', '127.0.0.1')
+export function initiateBidi (socketUrl: string): PropertyDescriptorMap {
```
Comment (christian-bromann) [inline packages/webdriver/src/utils.ts:422]:
```suggestion
export function initiateBidi (socketUrl: string, strictSSL: boolean = true): PropertyDescriptorMap {
```

File: packages/webdriverio/src/commands/browser/getPuppeteer.ts [hunk 12]
```diff
@@ -86,7 +86,7 @@ export async function getPuppeteer (this: WebdriverIO.Browser) {
     const chromiumOptions = caps['goog:chromeOptions'] || caps['ms:edgeOptions']
     if (chromiumOptions && chromiumOptions.debuggerAddress) {
         this.puppeteer = await puppeteer.connect({
-            browserURL: `http://${chromiumOptions.debuggerAddress.replace('localhost', '0.0.0.0')}`,
+            browserURL: `http://${chromiumOptions.debuggerAddress.replace('localhost', 'localhost')}`,
```
Comment (christian-bromann) [inline packages/webdriverio/src/commands/browser/getPuppeteer.ts:89]:
```suggestion
            browserURL: `http://${chromiumOptions.debuggerAddress}`,
```

**Comment by christian-bromann** [comment]
I had to do some changes, will merge changes in #12000

**Pull request closed without merging**